src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -270,6 +270,94 @@ class RgwService(CephadmService):
     TYPE = 'rgw'
 
     def config(self, spec: RGWSpec):
+                # check cluster health
+        ret, out, err = self.mgr.check_mon_command({
+            'prefix': 'health',
+            'format': 'json',
+        })
+        try:
+            j = json.loads(out)
+            self.mgr.log.info(j)
+        except Exception as e:
+            raise OrchestratorError('failed to parse health status')
+
+        if j['status'] != 'HEALTH_OK':
+                raise OrchestratorError('Heath not ok')
+
+        # get realms
+        out, err, code = self.mgr._run_cephadm(self.mgr.cache.get_hosts()[0], 'client', 'shell', ['--', 'radosgw-admin', 'realm', 'list', '--format=json'], error_ok=True)
```

Comment:
```suggestion
        out, err, code = self.mgr._run_cephadm(self.mgr.cache.get_hosts()[0], spec.service_name(), 'shell', ['--', 'radosgw-admin', 'realm', 'list', '--format=json'], error_ok=True)
```

the entity param should be set to `rgw` not `client`